Sources/Orbit/Components/Slider.swift

```diff
@@ -0,0 +1,421 @@
+        }
+    }
+
+    @ViewBuilder var track: some View {
+        RoundedRectangle(cornerRadius: 100)
+            .fill(Color.cloudDark)
+            .frame(maxHeight: lineHeight)
+    }
+
+    @ViewBuilder func controls(width: CGFloat) -> some View {
+
+        let firstThumbPosition = position(by: firstValue, sliderWidth: width)
+
+        switch secondValue {
+            case nil:
+                firstThumb(atPosition: firstThumbPosition, sliderWidth: width)
+                    .background(strokedPath(from: 0, to: firstThumbPosition))
+
+            case .some(let secondValue):
+                let secondThumbPosition = position(by: secondValue, sliderWidth: width)
+                ZStack {
+                    firstThumb(atPosition: firstThumbPosition, sliderWidth: width)
+                    secondThumb(atPosition: secondThumbPosition, sliderWidth: width)
+                }
+                .background(strokedPath(from: firstThumbPosition, to: secondThumbPosition))
+        }
+    }
+
+    @ViewBuilder func firstThumb(atPosition horisontalPosition: CGFloat, sliderWidth: CGFloat) -> some View {
+        thumb(
+            horisontalPosition: horisontalPosition,
+            sliderWidth: sliderWidth,
+            isDragging: $isFirstThumbDrag
+        ) {
+            self.firstValue = $0
+        }
+        .zIndex(isFirstThumbDrag ? 1 : 0)
+    }
+
+    @ViewBuilder func secondThumb(atPosition horisontalPosition: CGFloat, sliderWidth: CGFloat) -> some View {
+        thumb(
+            horisontalPosition: horisontalPosition,
+            sliderWidth: sliderWidth,
+            isDragging: $isSecondThumbDrag
+        ) {
+            self.secondValue = $0
+        }
+        .zIndex(isSecondThumbDrag ? 1 : 0)
+
+    }
+
+    @ViewBuilder func thumb(
+        horisontalPosition: CGFloat,
+        sliderWidth: CGFloat,
+        isDragging: Binding<Bool>,
+        onValueChange: @escaping (Value) -> Void
+    ) -> some View {
+        SliderThumb()
+            .position(x: horisontalPosition, y: lineCenter)
+            .highPriorityGesture(
+                DragGesture(minimumDistance: 1)
+                    .onChanged { value in
+
+                        if isDragging.wrappedValue == false {
+                            dragStartOffset = value.startLocation.x - horisontalPosition
+                            isDragging.wrappedValue = true
+                        }
+
+                        let newHorisontalPosition = value.location.x - dragStartOffset
+
+                        guard newHorisontalPosition < sliderWidth else {
+                            onValueChange(bounds.upperBound)
+                            return
+                        }
+
+                        guard newHorisontalPosition > 0 else {
+                            onValueChange(bounds.lowerBound)
+                            return
+                        }
+
+                        if step != nil {
```

Comment:
`step` isn't used in this `if` - it seems disconnected - why is `step` the deciding factor here? Shouldn't we use `buckets.isEmpty == false` or something like that?